k2/csrc/host/intersect.cc

```diff
@@ -53,7 +53,7 @@ void Intersection::GetSizes(Array2Size<int32_t> *fsa_size) {
   arc_map_b_.clear();
 
   if (IsEmpty(a_) || IsEmpty(b_)) return;
-  if (!IsArcSorted(a_) || !IsArcSorted(b_)) {
+  if (!check_properties_ || !IsArcSorted(a_) || !IsArcSorted(b_)) {
```

Comment:
Not a request for changing code.

I feel negative conditions are hard to read.

Reply:
I think it should be 

`if (check_properties_ && (!IsArcSorted(a_) || !IsArcSorted(b_)))`